src/main/web/CMakeLists.txt

```
@@ -2,21 +2,26 @@ set(SCRIPT "${PROJECT_SOURCE_DIR}/scripts/gen_embedded.py")
 
 
 set(EMBEDDED_FILES
-    ${CMAKE_CURRENT_SOURCE_DIR}/_404.png
-    ${CMAKE_CURRENT_SOURCE_DIR}/_error.css
-    ${CMAKE_CURRENT_SOURCE_DIR}/_error.html
-    ${CMAKE_CURRENT_SOURCE_DIR}/favicon.ico
-    ${CMAKE_CURRENT_SOURCE_DIR}/_jquery.min.js
-    ${CMAKE_CURRENT_SOURCE_DIR}/_seasocks.css
-    ${CMAKE_CURRENT_SOURCE_DIR}/_stats.html
-)
+        ${CMAKE_CURRENT_SOURCE_DIR}/_404.png
+        ${CMAKE_CURRENT_SOURCE_DIR}/_error.css
+        ${CMAKE_CURRENT_SOURCE_DIR}/_error.html
+        ${CMAKE_CURRENT_SOURCE_DIR}/favicon.ico
+        ${CMAKE_CURRENT_SOURCE_DIR}/_jquery.min.js
+        ${CMAKE_CURRENT_SOURCE_DIR}/_seasocks.css
+        ${CMAKE_CURRENT_SOURCE_DIR}/_stats.html
+        )
 
 
 add_custom_command(OUTPUT Embedded.cpp
-    COMMAND "${PYTHON_EXECUTABLE}" "${SCRIPT}" -o Embedded.cpp -f ${EMBEDDED_FILES}
-    COMMENT "Generating embedded content"
-    DEPENDS "${SCRIPT}"
-)
+        COMMAND "${PYTHON_EXECUTABLE}" "${SCRIPT}" -o Embedded.cpp -f ${EMBEDDED_FILES}
+        COMMENT "Generating embedded content"
+        DEPENDS "${SCRIPT}"
+        )
 add_library(embedded OBJECT Embedded.cpp)
 target_include_directories(embedded PUBLIC $<TARGET_PROPERTY:seasocks_headers,INTERFACE_INCLUDE_DIRECTORIES>)
+# TODO; for the life of me I don't understand why this if() doesn't work and we have to always build PIC.
+# symptom is a failed link building libseasocks.so only in conan.
+# repro case: `mkdir build; cd build; conan install ..; conan build ..`
+#if (SEASOCKS_SHARED)
 set_property(TARGET embedded PROPERTY POSITION_INDEPENDENT_CODE TRUE)
+#endif ()
```

I think it is working now?
```suggestion
if (SEASOCKS_SHARED)
    set_property(TARGET embedded PROPERTY POSITION_INDEPENDENT_CODE TRUE)
endif ()
```